snuba_sdk/request.py

```diff
@@ -81,9 +81,9 @@ def to_dict(self) -> dict[str, object]:
             serialized_mql = self.query.serialize()
             assert isinstance(serialized_mql, dict)  # mypy
             mql_context = serialized_mql["mql_context"]
-            query = str(serialized_mql["mql"])
+            query: Union[str, Dict[str, Any]] = str(serialized_mql["mql"])
         else:
-            query = str(self.query.serialize())
```

Comment:
I verified that this change doesn't have any logical consequences. It was only there before to appease mypy.

Reply:
With this change `ret` on line 88 can have a key `"query"` with type `str` if it's a metrics query. The same key will have a type `dict` if it is not a metric query. Is this an intentional outcome? I wonder if it'd create surprises in serialization logic. Also, I would appreciate a bit more context on what motivated this change in the PR description. Thanks!

Reply:
Hi Onkar thanks for taking the time to review this. Sorry for the bad description I wasnt expecting people without context to look at it. I added more context to the description so hopefully it makes more sense.

The PR has been changed slightly so now the type mismatch is not a thing anymore